integreat_cms/cms/utils/repair_tree.py

```diff
@@ -0,0 +1,284 @@
+
+    @write.setter
+    def write(self, new: Callable[[str], None]) -> None:
+        """
+        Set write function w/o new line
+        """
+        self._write = new
+
+
+@tree_mutex("page")
+def repair_tree(
+    page_id: int = 0, commit: bool = False, printer: Printer = Printer()
+) -> None:
+    """
+    Fix the tree for a given page or all trees.
+    """
+    mptt_fixer = MPTTFixer()
+
+    if page_id:
+        try:
+            page = Page.objects.get(id=page_id)
+        except Page.DoesNotExist as e:
+            raise ValueError(f'The page with id "{page_id}" does not exist.') from e
+        root_nodes = [mptt_fixer.get_fixed_root_node(page_id)]
+    else:
+        root_nodes = mptt_fixer.get_fixed_root_nodes()
+
+    for root_node in root_nodes:
+        action = "Fixing" if commit else "Detecting problems in"
+        printer.print("%s tree with id {root_node.tree_id}...", action)
+        for tree_node in mptt_fixer.get_fixed_tree_of_page(root_node.pk):
+            print_changed_fields(
+                Page.objects.get(id=tree_node.pk), tree_node.lft, tree_node.rgt, printer
+            )
+
+    if commit:
+        for page in mptt_fixer.get_fixed_tree_nodes():
+            page.save()
+
+
+def print_changed_fields(
+    tree_node: Page, left: int, right: int, printer: Printer = Printer()
+) -> None:
+    """
+    Check whether the tree fields are correct
+    """
+    printer.write(printer.bold(f"Page {tree_node.id}:"))
+    printer.success("\tparent_id: %i", tree_node.parent_id)
+    if not tree_node.parent or tree_node.tree_id == tree_node.parent.tree_id:
+        printer.success("\ttree_id: {tree_node.tree_id}")
+    else:
+        printer.error("\ttree_id: %i → %i", tree_node.tree_id, tree_node.parent.tree_id)
+    if tree_node.parent_id:
+        if tree_node.depth == tree_node.parent.depth + 1:
+            printer.success("\tdepth: %i", tree_node.depth)
+        else:
+            printer.error(
+                "\tdepth: %i → %i", tree_node.depth, tree_node.parent.depth + 1
+            )
+    elif tree_node.depth == 1:
+        printer.success("\tdepth: %i", tree_node.depth)
+    else:
+        printer.error("\tdepth: %i → 1", tree_node.depth)
+    if tree_node.lft == left:
+        printer.success("\tlft: %i", tree_node.lft)
+    else:
+        printer.error("\tlft: %i → %i", tree_node.lft, left)
+    if tree_node.rgt == right:
+        printer.success("\trgt: %i", tree_node.rgt)
+    else:
+        printer.error("\trgt: %i → %i", tree_node.rgt, right)
+
+
+class MPTTFixer:
+    """
+    Gets ALL nodes and coughs out fixed LFT, RGT and depth values. Uses the parent field
+    to fix hierarchy and sorts siblings by (potentially inconsistent) lft.
+    """
+
+    def __init__(self) -> None:
+        """
+        Creates a fixed tree when initializing class but does not save results
+        """
+        self.broken_nodes: list[Page] = list(
+            Page.objects.all().order_by("tree_id", "lft")
+        )
```

Comment:
```suggestion
        self.broken_root_nodes: list[Page] = list(
            Page.objects.filter(parent=None)
        )
        self.broken_child_nodes: list[Page] = list(
            Page.objects.exclude(parent=None)
        )
```
If those subsets are only handled separately, I think it makes sense to store them independent of each other instead of using the loops later.
Also, the ordering is like this by default, so no need to state it explicitly.